internal/util/pidlimit.go

```diff
@@ -78,7 +79,7 @@ func GetPIDLimit() (int, error) {
 	defer f.Close() // #nosec: error on close is not critical here
 
 	maxPidsStr, err := bufio.NewReader(f).ReadString('\n')
-	if err != nil && err != io.EOF {
+	if !errors.Is(err, nil) && !errors.Is(err, io.EOF) {
```

Comment:
`!errors.Is(err, nil)` should just be `err != nil` :wink:

Checking `io.EOF` is an exception, I think. It does not need `errors.Is`. Did the linter really complain about this?

Reply:
`internal/util/pidlimit.go:82:29: comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error (errorlint)
	if !errors.Is(err, nil) && err != io.EOF {`

Reply:
hmm, ok. But still need to change back to `err != nil`, as `nil` is not a wrapper error.

Reply:
```suggestion
if err != nil{
    if  !errors.Is(err, io.EOF) {
      return 0, err
    }
}

Reply:
`if err != nil && !errors.Is(err, io.EOF) ` was working. Hence changed to that only.